### PR TITLE
created the performance subsystem

### PR DESCRIPTION
**Pull request by robsim378**
- implemented performance measurement gathering.
- for performance calculations, the csv output files can be imported into excel

### CODE REVIEW

File: src/sysc_3303_project/floor_subsystem/InputFileController.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 import sysc_3303_project.common.events.DelayTimerThread;
 import sysc_3303_project.common.events.Event;
 import sysc_3303_project.common.events.RequestData;
+import sysc_3303_project.performance_tester.PerformanceEventType;
```
Comment (dwarflord1) [inline src/sysc_3303_project/floor_subsystem/InputFileController.java:20]: Unnessesary import

File: src/sysc_3303_project/performance_tester/PerformanceTester.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+     * @return  PerformanceRequestData, the first request made to that floor.
+     */
+    private PerformanceRequestData findPendingRequest(int destinationFloor) {
+        for (PerformanceRequestData i : pendingRequests) {
+            if (i.getDestinationFloor() == destinationFloor) {
+                return i;
+            }
+        }
+        return null;
+    }
```
Comment (dwarflord1) [inline src/sysc_3303_project/performance_tester/PerformanceTester.java:63]: Finding the first destination floor that matches seems like a recipe for errors if multiple requests end on the same space. Is this accounted for anywhere or are we just going to not test these cases?

File: src/sysc_3303_project/performance_tester/PerformanceTester.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+     * List of requests that have been read from the input file but not yet scheduled
+     */
+    private ArrayList<PerformanceRequestData> pendingRequests;
+
+    public PerformanceTester(EventBuffer<PerformanceEventType> inputBuffer) {
+        this.inputBuffer = inputBuffer;
+
+    }
+
+    private void outputToFile(String filePath, PerformanceRequestData data, LocalTime servicedTime) {
```
Comment (dwarflord1) [inline src/sysc_3303_project/performance_tester/PerformanceTester.java:47]: Will this be implemented later?

File: src/sysc_3303_project/performance_tester/PerformanceTester.java (hunk 2 above)
Comment (ianh6) [reply src/sysc_3303_project/performance_tester/PerformanceTester.java:63]: this is accounted for in the ongoingRequestServiced() method

File: src/sysc_3303_project/performance_tester/PerformanceTester.java (hunk 3 above)
Comment (ianh6) [reply src/sysc_3303_project/performance_tester/PerformanceTester.java:47]: I will be implementing this today

**Review by Andrei486 — APPROVED** [review]
note: will break tests

**Pull request merged**